### PR TITLE
feat: add option to hide label when battery is full

### PR DESCRIPTION
**Pull request by rubiin**
closes https://github.com/Jas-SinghFSU/HyprPanel/issues/313

Adds an option to toggle the battery label on bar when the charge is full

image

image

### CODE REVIEW

**Comment by Jas-SinghFSU** [comment]
:eyes: On a roll lol.

File: modules/bar/battery/index.ts [hunk 1]
```diff
@@ -59,29 +59,41 @@ const BatteryLabel = (): BarBoxChild => {
...
-                }
-            }),
+            children: Utils.merge(
+                [
+                    battery.bind('available'),
+                    show_label.bind('value'),
+                    battery.bind('charged'),
+                    hideLabelWhenFull.bind('value'),
+                ],
+                (batAvail, showLabel, hideWhenFull, isCharged) => {
```
Comment (Jas-SinghFSU) [inline modules/bar/battery/index.ts:69]:
```suggestion
                    hideLabelWhenFull.bind('value'),
                ],
                (batAvail, showLabel, isCharged, hideWhenFull) => {
```
These two are swapped.

**Review by Jas-SinghFSU — CHANGES_REQUESTED** [review]

**Comment by Jas-SinghFSU** [comment]
Similarly, if the label is hidden, the split style icon needs to have it border adjusted. We can add the `last-child` scope at the bottom of this css.
```ts
    .bar-button-icon.battery {
        border-top-left-radius: $bar-buttons-radius;
        border-bottom-left-radius: $bar-buttons-radius;
        background: if($bar-buttons-monochrome, $bar-buttons-icon, $bar-buttons-battery-icon_background);
        padding: $bar-buttons-padding_y 0em;
        padding-left: $bar-buttons-padding_x;
        padding-right: $bar-buttons-battery-spacing;
        border-top-left-radius: if(
            $bar-buttons-battery-enableBorder or $bar-buttons-enableBorders,
            $bar-buttons-radius * 0.4,
            $bar-buttons-radius
        );
        border-bottom-left-radius: if(
            $bar-buttons-battery-enableBorder or $bar-buttons-enableBorders,
            $bar-buttons-radius * 0.4,
            $bar-buttons-radius
        );

        &:last-child {
            border-radius: $bar-buttons-radius;
        }
    }
```

**Comment by rubiin** [comment]
> 👀 On a roll lol.

Rick Astley Never Gonna Give You Up GIF – Rick Astley Never Gonna Give You Up Rickroll – discover and share GIFs

**Review by Jas-SinghFSU — APPROVED** [review]

**Pull request merged**